commons/zenoh-config/src/lib.rs

```diff
@@ -100,6 +100,52 @@ pub struct DownsamplingItemConf {
     pub flow: DownsamplingFlow,
 }
 
+//adding datatypes needed for ACL Config
+
+#[derive(Serialize, Debug, Deserialize, Clone)]
+
+pub struct ConfigRule {
+    pub interface: Vec<String>,
+    pub key_expr: Vec<String>,
+    pub action: Vec<Action>,
+    pub permission: Permission,
+}
+#[derive(Clone, Serialize, Debug, Deserialize)]
+pub struct PolicyRule {
+    pub subject: Subject, //Subject
+    pub key_expr: String,
+    pub action: Action,         //Action
+    pub permission: Permission, //Permission
+}
+
+#[derive(Serialize, Debug, Deserialize, Eq, PartialEq, Hash, Clone)]
+#[serde(untagged)]
+#[serde(rename_all = "snake_case")]
+
+pub enum Subject {
+    Interface(String),
+    //Username(String)
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, Hash, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum Action {
+    Put,
+    DeclareSubscriber,
+    Get,
+    DeclareQueryable,
+}
+pub const NUMBER_OF_ACTIONS: usize = 4; //size of Action enum (change according to Action size)
```

Comment:
I think this `const` is actually not necessary because of its usage in other parts of the code.